lib/rules/no-implied-eval.js

```diff
@@ -28,141 +34,131 @@ module.exports = {
+        /**
+         * Reports calls of `implied eval` via the global references.
+         * @param {Variable} globalVar A global variable to check.
+         * @returns {void}
+         */
+        function reportImpliedEvalViaGlobal(globalVar) {
+            const { references, name } = globalVar;
 
-        return {
-            CallExpression(node) {
-                if (isImpliedEvalCallExpression(node)) {
+            references.forEach(ref => {
+                const identifier = ref.identifier;
+                let node = identifier.parent;
 
-                    // call expressions create a new substack
-                    impliedEvalAncestorsStack.push([node]);
+                while (isSpecifiedMember(node, [name])) {
+                    node = node.parent;
                 }
-            },
-
-            "CallExpression:exit"(node) {
-                if (node === last(last(impliedEvalAncestorsStack))) {
 
-                    /*
-                     * Destroys the entire sub-stack, rather than just using
-                     * last(impliedEvalAncestorsStack).pop(), as a CallExpression is
-                     * always the bottom of a impliedEvalAncestorsStack substack.
-                     */
-                    impliedEvalAncestorsStack.pop();
+                if (isSpecifiedMember(node, EVAL_LIKE_FUNCS)) {
+                    node = node.parent;
+                    reportImpliedEvalCallExpression(node);
                 }
-            },
+            });
+        }
 
-            BinaryExpression(node) {
-                if (node.operator === "+" && hasImpliedEvalParent(node)) {
-                    last(impliedEvalAncestorsStack).push(node);
-                }
-            },
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
 
-            "BinaryExpression:exit"(node) {
-                if (node === last(last(impliedEvalAncestorsStack))) {
-                    last(impliedEvalAncestorsStack).pop();
+        return {
+            CallExpression(node) {
+                if (isSpecifiedIdentifier(node.callee, EVAL_LIKE_FUNCS)) {
+                    reportImpliedEvalCallExpression(node);
                 }
             },
+            "Program:exit"() {
+                const globalScope = context.getScope();
 
-            Literal(node) {
-                if (typeof node.value === "string") {
-                    checkString(node);
-                }
-            },
+                const globalVariables = GLOBAL_CANDIDATES
+                    .map(candidate => astUtils.getVariableByName(globalScope, candidate))
+                    .filter(globalVar => !!globalVar);
 
-            TemplateLiteral(node) {
-                checkString(node);
+                globalVariables.forEach(reportImpliedEvalViaGlobal);
```

Comment:
Consider adding this to the method chain above since I don't think the temp `globalVariables` is necessary here (I think the code is fairly easy to read!).

Reply:
@kaicataldo I agree :)